### DISCUSSION TITLE
Running locally

### DISCUSSION THREAD

**Discussion by rolanddosa** [electricitymaps/electricitymaps-contrib#2475] [General]
Hi all, I am trying to do 'docker-compose build' on the master branch, but I have this issue: 

**FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory:**
     I tried increasing the space with     `"build-release": "export NODE_ENV=production && NODE_OPTIONS=--max-old-space-size=8192 && webpack --bail --mode production",`   but still failing..

Anyone else had this issue? Let me know any suggestions please, thank you

**Comment by fbarl** [comment]
Hi @rolanddosa, thanks for reporting the issue!

Could you please tell us which OS you're using and share the output of the following commands to try to reproduce your problem:

```
node --version
docker --version
docker-compose --version
```

My bet is that upgrading your node [might fix it](https://github.com/webpack/webpack-sources/issues/66#issuecomment-531862216), that's what I'd try next!

**Comment by rolanddosa** [comment]
Hi just upgraded my node, but still no luck..

I'm running on Windows. 
$ node --version
v12.16.3
$ docker --version
Docker version 19.03.1, build 74b1e89e8a
$ docker-compose --version
docker-compose version 1.24.1, build 4667896b

Also I had to get Docker Tools as I have Windows 10 Home, would that be an issue?

**Comment by rolanddosa** [comment]
Fixed it, the syntax was wrong, it worked with     `"build-release": "export NODE_OPTIONS=--max-old-space-size=4096 NODE_ENV=production && webpack --bail --mode production",`

The other issue I'm facing now is  my node is stuck on starting:
```
web-watch_1   | <s> [webpack.Progress] 98% after emitting
web-watch_1   | <s> [webpack.Progress] 100%
web_1         | [nodemon] starting `node server.js`
web-watch_1   |
web_1         | [nodemon] restarting due to changes...
web_1         | [nodemon] starting `node server.js`



```

**Comment by rolanddosa** [comment]
Fixed now, I was checking localhost instead of the Docker's IP address. Will close this issue now.

**Comment by fbarl** [comment]
Glad it all worked out for you in the end @rolanddosa, and thanks for sharing all the bumps you experienced along the road :wink:

**Comment by andy-jdl** [comment]
Hi I'm new here. I wanted to get started with front end development and I am trying to run the mockserver locally. I followed the contribution guidelines and when I go to localhost:8001 I receive a GET error. Is anyone familiar with this?

Running Windows
-node 20.3.0

Greatly appreciate any help

**Reply by VIKTORVAV99** [reply]
Hi!

Just to clear things up a little.
You should be running the mockserver in one terminal and the frontend in another.
This is done by running `pnpm run mockserver` in the web directory and then opening a new terminal and running `pnpm run dev`, also in the web directory.

At this point a new browser window should open and you should see both the app interface as well as a colored map.

You should not be trying to access the mock server directly, it is just there to color the map instead of the full backend.

Also just to note, we have not upgraded to node 20 yet as it has not reached LTS so I don't know if it is working on that version, I am personally running the latest version of node 18 (LTS) and that works with the above instructions.
